libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1072,19 +1085,32 @@ class MapboxNavigation(
     }
 
     private fun restoreTripSessionRoute() {
-        val legIndex = latestLegIndex ?: directionsSession.initialLegIndex
-        tripSession.setRoute(directionsSession.routes.firstOrNull(), legIndex)
+        ifNonNull(latestDirectionsResponse) {
```

Comment:
had issues rebasing on this one https://github.com/mapbox/mapbox-navigation-android/pull/4892